### PR TITLE
Add pregenerated bindings

### LINKED ISSUE #23: Vendor the upstream repository for ethercat-sys
To make compiling `ethercat-sys` easier, is it possible to vendor `http://hg.code.sf.net/p/etherlabmaster/code` as a git submodule? 

The idea being that

- the `git clone ... && cargo build` workflow will *Just Work*, and it won't impact projects which transitively depend on `ethercat-sys`
- The docs.rs build will complete so you can read `ethercat`'s docs online ([build logs](https://docs.rs/crate/ethercat/0.1.3/builds/191051))
- It's not possible for someone to accidentally use a different checkout of the upstream repository
- You can remove [this](https://github.com/ethercat-rs/ethercat/blob/7584cca67de2e7793110bef5ace5cbd7b6d2579b/ethercat-sys/build.rs#L8-L9) panic

I know the original repository uses mercurial, but there seems to be [a way to bridge `git` and `mercurial`](https://stackoverflow.com/questions/9067283/is-there-a-way-to-use-a-mercurial-repository-as-git-submodule). 

When publishing to crates.io, `cargo` will just grab the files on disk and not go through git/mercurial. That means people using the published crate won't need to use tools like `git-hg`.

### PR DESCRIPTION
**Pull request by flosse**
this should fix #23

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,11 +31,19 @@ default = []
 # Enable this feature to use it with the
 # synapticon branch `release/v1.5.2-sncn-11`
 # at https://github.com/synapticon/Etherlab_EtherCAT_Master
-sncn = []
+sncn = ["ethercat-sys/sncn"]
+
+# Enable this feature to use pregenerated bindings.
+# CAUTION: If your kernel module was not build
```
Comment (birkenfeld) [inline Cargo.toml]: nit:
```suggestion
# CAUTION: If your kernel module was not built
```

File: ethercat-sys/build.rs [hunk 2]
```diff
@@ -4,87 +4,89 @@
 use std::{env, fmt::Write, fs, path::PathBuf};
 
 fn main() {
-    let path = env::var("ETHERCAT_PATH").expect(
-        "Please set the ETHERCAT_PATH env var to the location of \
+    if env::var(format!("CARGO_FEATURE_PREGENERATED_BINDINGS")).is_err() {
```
Comment (birkenfeld) [inline ethercat-sys/build.rs]: Why `format!`?

**Review by birkenfeld — APPROVED** [review]
Only minor points. Is there some automated process to regenerate these?

File: Cargo.toml (hunk 1 above)
Comment (flosse) [reply Cargo.toml]: thx:)

File: ethercat-sys/build.rs (hunk 2 above)
Comment (flosse) [reply ethercat-sys/build.rs]: That's why I copied it from an other build file that used variables ;-)

**Comment by flosse** [comment]
> Is there some automated process to regenerate these?

no. We could add a shell script to automatically copy & rename the generated files but I would not prioritize this yet. This could be done in e.g. a `v0.2.1` release.

**Pull request merged**